telemetry/test_convert.py

```diff
@@ -291,6 +291,86 @@ def test_normal(self):
             self.assertEqual(converted["histograms"][h], expected_converted_histograms[h])
         self.assertIs(converted["info"].get("geoCountry"), None)
 
+    def print_byte_range(self, data, start=None, end=None):
+        if start is None:
+            start = 0
+        if end is None:
+            end = len(data)
+        for i in range(start, end):
+            pieces = {
+                "i": i,
+                "o": ord(data[i]),
+                "c": data[i]
+            }
+            print "0x%(i)04x %(i)04d = 0x%(o)02x %(o)3d %(c)s" % pieces
+        print "----"
+
+    def test_utf8(self):
+        count = 0
+        # This packed file contains the same record twice, once gzipped, once
+        # raw. The record contains the UTF-8 encoded string "Wikipédia"
+        for r in fu.unpack('test/unicode.v1.packed', file_version="v1"):
+            self.assertIs(r.error, None)
+            self.assertTrue(len(r.data) > 0)
+            self.assertEqual(r.data[0], '{')
+            count += 1
+
+            # self.print_byte_range(r.data, 3323, 3362)
+
+            # Make sure that the raw bytes are correct:
+            # Incoming is UTF-8, so we expect
+            #   W i k i p 0xc3 0xc9 d i a
+            self.assertEqual(ord(r.data[0xd16]), 0x70)
+            self.assertEqual(ord(r.data[0xd17]), 0xc3)
+            self.assertEqual(ord(r.data[0xd18]), 0xa9)
+            self.assertEqual(ord(r.data[0xd19]), 0x64)
+
+            # Convert the data the wrong way:
+            bad = unicode(r.data, errors="replace")
+            # self.print_byte_range(bad, 3323, 3362)
+
+            # Verify that we see the replacement chars in the expected places:
+            #   W i k i p 0xfffd 0xfffd d i a
```

Comment:
Without specifying the encoding, the `unicode()` function assumes ASCII encoding :(

https://docs.python.org/2/howto/unicode.html#the-unicode-type